book/src/assets/how_to_use_assets.md

```diff
@@ -0,0 +1,221 @@
+# How to Use Assets
+
+This guide covers the basic usage of assets into Amethyst for existing supported formats. For a list of supported formats, please [use this search for "Format"][doc_search_format] in the API documentation, and filter by the following crates:
+
+* amethyst_assets
+* amethyst_audio
+* amethyst_gltf
+* amethyst_locale
+* amethyst_ui
+
+## Steps
+
+1. Instantiate the Amethyst application with the assets directory.
+
+    ```rust,no_run,noplaypen
+    # extern crate amethyst;
+    #
+    use amethyst::{
+        prelude::*,
+        utils::application_root_dir,
+    };
+    #
+    # pub struct LoadingState;
+    # impl SimpleState for LoadingState {}
+
+    fn main() -> amethyst::Result<()> {
+        // Sets up the application to read assets in
+        // `<app_dir>/assets`
+        let app_root = application_root_dir()?;
+        let assets_dir = app_root.join("assets");
+
+        //..
+    #   let game_data = GameDataBuilder::default();
+
+        let mut game = Application::new(assets_dir, LoadingState, game_data)?;
+    #
+    #   game.run();
+    #   Ok(())
+    }
+    ```
+
+2. Ensure that the [`Processor<A>` system][doc_processor_system] for asset type `A` is registered in the dispatcher.
+
+    For asset type `A`, `Processor<A>` is a `System` that will asynchronously load `A` assets. Usually the crate that provides `A` will also register `Processor<A>` through a `SystemBundle`. Examples:
+
+    * `FontAsset` is provided by `amethyst_ui`, `UiBundle` registers `Processor<FontAsset>`.
+    * `Source` is provided by `amethyst_audio`, `AudioBundle` registers `Processor<Source>`.
+    * `SpriteSheet` is provided by `amethyst_renderer`, `RenderBundle` registers `Processor<SpriteSheet>` given `.with_sprite_sheet_processor()` is called before adding it to the `GameDataBuilder`.
+
+    ```rust,no_run,noplaypen
+    # extern crate amethyst;
+    #
+    # use amethyst::{
+    #     prelude::*,
+    #     renderer::{DisplayConfig, Pipeline, RenderBundle, Stage},
+    #     ui::{DrawUi, UiBundle},
+    #     utils::application_root_dir,
+    # };
+    #
+    # pub struct LoadingState;
+    # impl SimpleState for LoadingState {}
+
+    fn main() -> amethyst::Result<()> {
+    #   let app_root = application_root_dir()?;
+    #   let assets_dir = app_root.join("assets");
+    #
+    #   let display_config = DisplayConfig::default();
+    #   let pipeline = Pipeline::build().with_stage(
+    #       Stage::with_backbuffer()
+    #           .clear_target([0.; 4], 0.)
+    #           .with_pass(DrawUi::new()),
+    #   );
+        // ..
+
+        let game_data = GameDataBuilder::default()
+            .with_bundle(UiBundle::<String, String>::new())?
+            .with_bundle(
+                RenderBundle::new(pipeline, Some(display_config))
+                    .with_sprite_sheet_processor()
```

Comment:
Given Sprites are covered separately, I think we should use a different type here.

Reply:
I could remove this one specifically, since `UiBundle` already demonstrates a bundle with a `Processor` &ndash; was thinking this is a good example for "sometimes you need to configure the bundle". Would you prefer this be removed?

Reply:
I don't think we need to worry about representing things that are outside the scope of the given section of the book, or overlap with other sections. Bundles is more of a bigger picture detail, not something for the scope of this section.

Reply:
Re:original question, do you think the `UiBundle` is enough, or what else would you suggest?

---

Bundles should be mentioned here though &ndash; imagine a reader who wants to use an asset:

1. I need to tell Amethyst where my assets live &ndash; assets directory. Easy enough.
2. Oh, the book says I need to tell Amethyst to process them, otherwise they are never loaded.
3. ...

In step 2, we could simply tell them "add the `Processor<A>`". However, Amethyst also provides these systems in bundles &ndash; it's a convention / common practice. If we do not also write this here, they wouldn't know to look (unless we link it, but we don't have somewhere to link it to yet), so I think it's valuable to keep.

Reply:
I think, then, this falls prey to your "let's not use code to explain code" point.